File: 2ndLifeFront/src/app/services/admin.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const adminUrl = 'http://localhost:8083';

@Injectable({
  providedIn: 'root'
})
export class AdminService {
  constructor(private httpClient: HttpClient) {}

  //Teacher_CRUD
  addTeacher(teacherDto: any): Observable<any> {
    return this.httpClient.post(adminUrl + "/api/admin/teacher", teacherDto);
  }

  getAllTeachers(): Observable<any>{
    return this.httpClient.get(adminUrl + "/api/admin/teachers");
  }

  deleteTeacher(id: number): Observable<any>{
    return this.httpClient.delete(adminUrl + "/api/admin/teacher/" + id);
  }

  getTeacherById(id: number): Observable<any>{
    return this.httpClient.get(adminUrl + "/api/admin/teacher/" + id);
  }

  updateTeacher(teacherId:number , teacherDto: any): Observable<any>{
    return this.httpClient.put(adminUrl + "/api/admin/teacher/" +  teacherId , teacherDto);
  }


  //Course_CRUD
  addCourse(courseDto: any): Observable<any> {
    return this.httpClient.post(adminUrl + "/api/admin/course", courseDto);
  }

  getAllCourses(): Observable<any>{
    return this.httpClient.get(adminUrl + "/api/admin/courses");
  }

  deleteCourse(id: number): Observable<any>{
    return this.httpClient.delete(adminUrl + "/api/admin/course/" + id);
  }

  getCourseById(id: number): Observable<any>{
    return this.httpClient.get(adminUrl + "/api/admin/course/" + id);
  }

  updateCourse(courseId:number , courseDto: any): Observable<any>{
    return this.httpClient.put(adminUrl + "/api/admin/course/" +  courseId , courseDto);
  }


  //ENROLLMENTS
  getCourseEnrollments(): Observable<any>{
    return this.httpClient.get(adminUrl + "/api/admin/course/enrollments");
  }

  changeEnrollmentStatus(enrollmentId:number, status:string): Observable<any>{
    return this.httpClient.get(`${adminUrl}/api/admin/course/enrollment/${enrollmentId}/${status}`);
  }
  
}

File: eduLine-Backend/src/main/java/com/example/perfectionnement/eduLine/services/admin/AdminService.java
package com.example.perfectionnement.eduLine.services.admin;

import java.io.IOException;
import java.util.List;

import com.example.perfectionnement.eduLine.dto.CourseDto;
import com.example.perfectionnement.eduLine.dto.EnrollCourseDto;
import com.example.perfectionnement.eduLine.dto.TeacherDto;


public interface AdminService {

	//TEACHER_CRUD
	boolean addTeacher(TeacherDto teacherDto) throws IOException;
	
	List<TeacherDto> getAllTeachers();
	
	void deleteTeacher(Long id);
	
	TeacherDto getTeacherById(Long id);
	
	boolean updateTeacher(Long teacherId, TeacherDto teacherDto) throws IOException;

	
	//COURSE_CRUD
	boolean addCourse(CourseDto courseDto) throws IOException;
	
	List<CourseDto> getAllCourses();
	
	void deleteCourse(Long id);
	
	CourseDto getCourseById(Long id);
	
	boolean updateCourse(Long courseId, CourseDto courseDto) throws IOException;
	
	List<EnrollCourseDto> getEnrollments();
	
	boolean changeEnrollmentStatus(Long enrollmentId, String status);
}
